src/object/cli_obj.c

```diff
@@ -4758,38 +4759,37 @@ obj_comp_cb(tse_task_t *task, void *data)
 
 		if (task->dt_result == -DER_CSUM || task->dt_result == -DER_TX_UNCERTAIN ||
 		    task->dt_result == -DER_NVME_IO) {
-			if (!obj_auxi->spec_shard && !obj_auxi->spec_group &&
-			    !obj_auxi->no_retry && !obj_auxi->ec_wait_recov) {
+			if (!obj_auxi->spec_shard && !obj_auxi->spec_group && !obj_auxi->no_retry &&
+			    !obj_auxi->ec_wait_recov) {
 				/* Retry fetch on alternative shard */
-				if (obj_auxi->opc == DAOS_OBJ_RPC_FETCH) {
-					if (task->dt_result == -DER_CSUM)
+				if ((obj_auxi->opc == DAOS_OBJ_RPC_FETCH ||
+				     obj_auxi->opc == DAOS_OBJ_RPC_UPDATE) &&
+				    task->dt_result == -DER_CSUM) {
```

Comment:
I thought we only want to retry for csum error on wire? but not for NVME fetch CSUM error? No?

Reply:
Yeah that's correct, but it would lead to a much large change, and we also want to get a confirmation from the DAOS community to make sure it's fine to move that forward.

Therefore, our strategy here is to land a quick fix while waiting for it.

Reply:
oh, ok